src/vm/threads.h

```diff
@@ -5297,6 +5301,20 @@ class Thread: public IUnknown
     {
         m_gcModeOnSuspension = 0;
     }
+
+    LPCGUID GetActivityId() const
+    {
+        LIMITED_METHOD_CONTRACT;
+        return &m_activityId;
+    }
+
+    void SetActivityId(LPCGUID pActivityId)
+    {
+        LIMITED_METHOD_CONTRACT;
+        _ASSERTE(pActivityId != NULL);
+
+        m_activityId = *pActivityId;
```

Comment:
If pActivityId is NULL in release builds, this is fail with a seg fault. (right?) Would it be better to check and simply do nothing or memset m_activityId to zero to avoid a crash?

Reply:
Yes, I grappled with this one a bit, but chose not to do anything here because it is the responsibility of the public (or calling) API to prepare arguments.  Because this is only used by managed code, and Guid is a struct, you're guaranteed that it won't be NULL.  Just trying to avoid check overhead in this path - I need to see what I can do to reduce this kind of overhead in the WriteEvent path as well, which will be used much more often.